tensorflow-ops/examples/Regression.hs

```diff
@@ -0,0 +1,46 @@
+-- | Simple linear regression example for the README.
+
+import Control.Monad (replicateM, replicateM_, zipWithM)
+import System.Random (randomIO)
+import Text.Printf (printf)
+
+import qualified TensorFlow.Core as TF
+import qualified TensorFlow.GenOps.Core as TF
+import qualified TensorFlow.Gradient as TF
+import qualified TensorFlow.Ops as TF
+
+main :: IO ()
+main = do
+    -- Generate data where `y = x*3 + 8`.
+    xData <- replicateM 100 randomIO
+    let yData = [x*3 + 8 | x <- xData]
+    -- Fit linear regression model.
+    (w, b) <- fit xData yData
+    printf "w=%v b=%v\n" w b  -- w=3.0001216 b=7.999931
+
+fit :: [Float] -> [Float] -> IO (Float, Float)
+fit xData yData = TF.runSession $ do
+    -- Create tensorflow constants for x and y.
+    let x = TF.vector xData
+        y = TF.vector yData
+    -- Create scalar variables for slope and intercept.
+    w <- TF.build (TF.initializedVariable 0)
+    b <- TF.build (TF.initializedVariable 0)
+    -- Define the loss function.
+    let yHat = (x `TF.mul` w) `TF.add` b
+        loss = TF.square (yHat `TF.sub` y)
+    -- Optimize with gradient descent.
+    trainStep <- TF.build (gradientDescent 0.001 loss [w, b])
+    replicateM_ 1000 (TF.run trainStep)
+    -- Return the learned parameters.
+    (TF.Scalar w', TF.Scalar b') <- TF.run (w, b)
+    return (w', b')
+
+gradientDescent :: Float
```

Comment:
It doesn't look like there is a way to include a source file into the markdown, so as to force them to be in sync. I've included the full source in the example now + turned it into a test.